editor/plugins/curve_editor_plugin.cpp

```diff
@@ -713,61 +713,45 @@ Vector2 CurveEdit::get_world_pos(Vector2 p_view_pos) const {
 }
 
 // Uses non-baked points, but takes advantage of ordered iteration to be faster.
-template <typename T>
-static void plot_curve_accurate(const Curve &curve, float step, Vector2 scaling, T plot_func) {
-	if (curve.get_point_count() <= 1) {
-		// Not enough points to make a curve, so it's just a straight line.
-		// The added tiny vectors make the drawn line stay exactly within the bounds in practice.
-		float y = curve.sample(0);
-		plot_func(Vector2(0, y) * scaling + Vector2(0.5, 0), Vector2(1.f, y) * scaling - Vector2(1.5, 0), true);
+void CurveEdit::plot_curve_accurate(float p_step, const Color &p_line_color, const Color &p_edge_line_color) {
+	if (curve->get_point_count() <= 1) { // Draw single line through entire plot
+		float y = curve->sample(0);
+		draw_line(get_view_pos(Vector2(0.f, y)), get_view_pos(Vector2(1.f, y)), p_line_color, LINE_WIDTH, true);
+		return;
+	}
 
-	} else {
-		Vector2 first_point = curve.get_point_position(0);
-		Vector2 last_point = curve.get_point_position(curve.get_point_count() - 1);
-
-		// Edge lines
-		plot_func(Vector2(0, first_point.y) * scaling + Vector2(0.5, 0), first_point * scaling, false);
-		plot_func(Vector2(Curve::MAX_X, last_point.y) * scaling - Vector2(1.5, 0), last_point * scaling, false);
-
-		// Draw section by section, so that we get maximum precision near points.
-		// It's an accurate representation, but slower than using the baked one.
-		for (int i = 1; i < curve.get_point_count(); ++i) {
-			Vector2 a = curve.get_point_position(i - 1);
-			Vector2 b = curve.get_point_position(i);
-
-			Vector2 pos = a;
-			Vector2 prev_pos = a;
-
-			float scaled_step = step / scaling.x;
-			float samples = (b.x - a.x) / scaled_step;
-
-			for (int j = 1; j < samples; j++) {
-				float x = j * scaled_step;
-				pos.x = a.x + x;
-				pos.y = curve.sample_local_nocheck(i - 1, x);
-				plot_func(prev_pos * scaling, pos * scaling, true);
-				prev_pos = pos;
-			}
+	Vector2 first_point = curve->get_point_position(0);
+	Vector2 last_point = curve->get_point_position(curve->get_point_count() - 1);
 
-			plot_func(prev_pos * scaling, b * scaling, true);
-		}
-	}
-}
+	// Transform pixels-per-step into curve domain
+	const float world_step_size = p_step * _world_to_view.affine_inverse().get_scale().x;
```

Comment:
Given that the transform has no rotation, I'm sure there's an easier/more efficient way to get this, but I don't believe it matters given this is called once per frame tops.

Reply:
`1.0 / world_to_view.get_scale().x`?

Reply:
I love smart people 🥰 I double-checked with `Math::is_equal_approx()` and yuuup, that totally works!

Reply:
Well, if you grow something by X, you gotta shrink it by 1/X. That's what you're doing here